internal/clients/stubs/gcp_stub.go

```diff
@@ -60,19 +84,32 @@ func (mock *GCPClientStub) Status(ctx context.Context) error {
 }
 
 func (mock *GCPClientStub) GetInstanceDescriptionByID(ctx context.Context, id, zone string) (*clients.InstanceDescription, error) {
-	return nil, nil
+	for _, instanceID := range mock.Instances {
+		if ptr.From(instanceID) == id {
+			instanceDesc := &clients.InstanceDescription{ID: id, PublicIPv4: fmt.Sprintf("10.0.0.%v", ipCounter)}
+			ipCounter = ipCounter + 1
+			return instanceDesc, nil
+		}
+	}
+	return nil, MissingInstanceIDErr
 }
 
 func (mock *GCPClientStub) ListLaunchTemplates(ctx context.Context) ([]*clients.LaunchTemplate, error) {
 	return nil, nil
 }
 
 func (mock *GCPClientStub) InsertInstances(ctx context.Context, params *clients.GCPInstanceParams, amount int64) ([]*string, *string, error) {
-	return nil, nil, NotImplementedErr
+	for i := 0; i < int(amount); i++ {
+		ID := fmt.Sprintf("300394200587658274%s", strconv.Itoa(idCounter))
```

Comment:
We wouldn't need the counter, if we use the lenght of the array as lzap pointed out to me in one PR :)

https://github.com/RHEnVision/provisioning-backend/blob/6615db0a2e5cf96cafff28859ac859b285923dff/internal/clients/stubs/azure_stub.go#L72

```suggestion
		ID := fmt.Sprintf("300394200587658274%s", strconv.Itoa(strconv.Itoa(len(mock.Instances)+1)))
```